rdbms/src/main/java/org/spine3/server/storage/jdbc/query/WriteQuery.java

```diff
@@ -48,13 +48,17 @@ public void execute() {
                 statement.execute();
                 connection.commit();
             } catch (SQLException e) {
-                this.getLogger().error("Failed to execute write operation.", e);
+                getLogger().error("Failed to execute write operation.", e);
                 connection.rollback();
                 throw new DatabaseException(e);
             }
         }
     }
 
+    protected void logWriteError(Object id, SQLException e) {
+        getLogger().error("Failed to write record with id " + id, e);
```

Comment:
There is no overload for the `Logger#error` which would take `(String message, Object argument, Throwable error)`. That's why I used this notation to ensure that the whole error message (with the stack trace) is printed to the user.